src/runner.ts

```diff
@@ -44,7 +47,12 @@ export class ResolveRunner implements Types.IResolveRunner {
 
   private _source: any;
 
-  constructor(source: any, opts: Types.IResolveRunnerOpts = {}) {
+  constructor(
+    source: any,
+    opts: Types.IResolveRunnerOpts = {
+      graph: new DepGraph<string>({ circular: true }),
```

Comment:
Since runner never really stands on its own, I suggest requiring this, rather than having it be optional and create it's own graph.

Reply:
Just make it a new argument(source, graph, opts)?

Reply:
Removing the `new DepGraph` default altogether. `graph` is a required property, so it should never be needed.